### PR TITLE
Adds a wrapper bootstrap.php file to core's.

### LINKED ISSUE #32: Autoloading for tests in ExistingSite namespace does not work with core's bootstrap.php
**Describe the bug**
Using core's `bootstrap.php`, autoloading is not setup correctly for `foo/tests/src/ExistingSite` (and the corresponding JS one).

**To Reproduce**
* Add a base class that extends this project's base class, in module_a.
* In module_b, add a test that extends that class
* Note the base class in module_a is not found and the test fatal-errors.

**Expected behavior**
We need to provide an example bootstrap.php file that is capable of running the 2 new types of tests alongside the existing 4 core test types.

### PR DESCRIPTION
**Pull request by jhedstrom**
- Fixes #32

### CODE REVIEW

**Comment by weitzman** [comment]
If I'm reading this right, this adds even more directory scanning. Considering how slow Docker's filesystem is on a Mac, this will cause some groaning on my projects at least. 

I'm leaving the actual review and commit of this to @rbayliss as he has also dealt with this on our mass.gov project.

**Comment by weitzman** [comment]
Also needs a change to README.md and ideally some instructions for folks who already have a bootstrap.php and are now adopting DTT.

**Comment by rbayliss** [comment]
Thanks for the reminder to look at this one, @weitzman. On one hand, re-using core's bootstrap.php file does cause a duplicate file scan.  On the other, to fix that issue we'd have to duplicate the logic in `bootstrap.php`, which could cause incompatibilities between Drupal versions.  So I think this is the best way this problem can be solved in the library, and if people need a faster startup, they can create and manage their own bootstrap file.  

I really wish core just used registered a PSR-4 root at `$module/tests/src`, so maybe we can follow up with a core patch to add that.  If anyone's interested in pursuing it, here's some [background reading](https://www.drupal.org/node/2605664).

Anyway, long story short, this PR looks good to me, except that it needs:
* Documentation on how to use the bootstrap file.
* Documentation on what to do if you already have your own bootstrap file.
* Code style cleanup.

**Comment by weitzman** [comment]
So, drupal core does a lot lot file scanning because it is dealing with an uninstalled site. With DTT, we have an installed site. So we actually could just ask Drupal for the enabled modules list. Should take about 1 second - maybe more on docker for mac. Thoughts?

**Comment by jibran** [comment]
> So we actually could just ask Drupal for the enabled modules list.
This is tricky, it means initializing the container in `bootstrap.php` which is worse than the current solution not that I like it very much but we have no other choice to scan all the test directories.

**Comment by jibran** [comment]
I liked the old bootstrap.php in core better before all the suite classes stuff. @larowlan modified it for drupal project like this https://gist.github.com/jibran/546370505dab0cb547d89f0a2ea3d3c5

**Comment by rbayliss** [comment]
Yeah, I agree on not wanting to bootstrap Drupal for this.  We're just trying to populate the classloader here, and bootstrapping has side effects.  @jibran - I agree that implementation looks nice, but it is out of date with the bootstrap file in core (missing class aliases for PHPUnit v6 for example).  If we don't build off of the core bootstrap file, we have to try to sync our changes with the core version, which seems like it could cause trouble.

**Comment by jhedstrom** [comment]
I added some documentation to the `README.md`.

**Comment by jhedstrom** [comment]
Regarding the double scanning, since core's `bootstrap.php` file has this conditional:

```php
  if (!isset($GLOBALS['namespaces'])) {
    // Scan for arbitrary extension namespaces from core and contrib.
    $extension_roots = drupal_phpunit_contrib_extension_directory_roots();

    $dirs = array_map('drupal_phpunit_find_extension_directories', $extension_roots);
    $dirs = array_reduce($dirs, 'array_merge', []);
    $GLOBALS['namespaces'] = drupal_phpunit_get_extension_namespaces($dirs);
  }
```
It will not execute that since we set `$GLOBALS['namespaces']` in our own `bootstrap.php`.

**Comment by rbayliss** [comment]
Just tested and you're right @jhedstrom, there is no duplicate file scan.  This looks great.  Thanks!
